### ISSUE TITLE
Use specific exceptions instead of "except Exception"

### ISSUE THREAD

**Issue by jpmckinney** [open-contracting/kingfisher-colab#24]
Only one occurrence remains (below).

**Comment by jpmckinney** [comment]
@kindly @duncandewhurst `reset_connection` is:

```python
global conn
if conn and not conn.closed:
    try:
        conn.cancel()
        conn.reset()
    except Exception:
        pass

conn = None
```

A few questions:

* What is the use case for resetting the connection?
* Why [cancel()](https://www.psycopg.org/docs/connection.html#connection.cancel)? Is it for a user to deliberately interrupt a slow query? In that case, is the only exception we're trying to catch `psycopg2.extensions.QueryCanceledError`?
* Why [reset()](https://www.psycopg.org/docs/connection.html#connection.reset)? We set the connection to `None` immediately after. Do we instead mean to `conn.close()` (which we should call anyhow)?

**Comment by jpmckinney** [comment]
This seems to only be used in old notebooks, where there was this content:

> Sometimes you'll get a strange error saying the cursor is already closed. If this persists after refreshing the colab notebook run the following cell, then re-run the previous cell to connect again

**Comment by jpmckinney** [comment]
Closed by f8be196
